### ISSUE TITLE
Gitlens don't refresh remote branch deleted with ssh remote

### ISSUE THREAD

**Issue by Jedore** [gitkraken/vscode-gitlens#1568]
- GitLens Version: 11.5.1
- Git Version: 2.30.1
- VSCode Version: 1.57.1 (user setup)
- OS: Windows_NT x64 10.0.19042
- Remote OS: CentOS 7.9

I develop with ssh remote CentOS.  There is a branch `feature-admin`:
```bash
$ git branch -a
  develop
* feature-admin
  remotes/origin/HEAD -> origin/develop
  remotes/origin/develop
  remotes/origin/feature-admin
  remotes/origin/feature-register
  remotes/origin/master
  remotes/origin/old-develop
  remotes/origin/release-0.1
  remotes/origin/test
```
then I deleted some remote branch in browser：
```bash
$ git remote show origin
* remote origin
  Fetch URL: git@gitee.com:yuezhix/knowledge_work.git
  Push  URL: git@gitee.com:yuezhix/knowledge_work.git
  HEAD branch: develop
  Remote branches:
    develop                           tracked
    feature-register                  tracked
    master                            tracked
    old-develop                       tracked
    refs/remotes/origin/feature-admin stale (use 'git remote prune' to remove)
    release-0.1                       tracked
    test                              tracked
  Local branches configured for 'git pull':
    develop       merges with remote develop
    feature-admin merges with remote feature-admin
  Local ref configured for 'git push':
    develop pushes to develop (local out of date)
```
but `feature-admin` still exists locally after I refresh or fetch Gitlens's remotes menu:
```bash
$ git branch -a
  develop
* feature-admin
  remotes/origin/HEAD -> origin/develop
  remotes/origin/develop
  remotes/origin/feature-admin
  remotes/origin/feature-register
  remotes/origin/master
  remotes/origin/old-develop
  remotes/origin/release-0.1
  remotes/origin/test
```
I find `feature-admin` will be removed by below command:
```bash
$ git remote prune origin
Pruning origin
URL: git@gitee.com:yuezhix/knowledge_work.git
 * [pruned] origin/feature-admin
$ git branch -a
  develop
* feature-admin
  remotes/origin/HEAD -> origin/develop
  remotes/origin/develop
  remotes/origin/feature-register
  remotes/origin/master
  remotes/origin/old-develop
  remotes/origin/release-0.1
  remotes/origin/test
```

So,  I think Gitlens should remove remote branch after it has been deleted in other place, does Gitlens has this feature ?

**Comment by yzayak2s** [comment]
I also faced the same problem. When I remove a branch it is still showing in the remotes after reloading (GitLens).

**Comment by jcbianic** [comment]
I also faced the same behaviour. I believe it has nothing to do with gitlens but relates to the underlying behavior of git fetch.
I stumbe upon [this StackOverflow Q&A](https://stackoverflow.com/questions/38512124/visual-studio-code-remove-branches-deleted-on-github-that-still-show-in-vs-cod/44854800#44854800) and the simple solution is to run  `git fetch --prune`. I tested it and it works like a charm. You don't even need to reload VSCode.

Digging up the git documentation on the `--prune` option [here](https://git-scm.com/docs/git-fetch#Documentation/git-fetch.txt--p),  it implies that without it fetch does not _remove any remote-tracking references that no longer exist on the remote_.
